components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/internal/ApplicationManagementServiceComponent.java

```diff
@@ -458,4 +459,28 @@ protected void unsetSAMLSSOServiceProviderManager(SAMLSSOServiceProviderManager
             log.debug("SAMLSSOServiceProviderManager unset in to bundle");
         }
     }
+
+    @Reference(
+            name = "identity.event.service",
+            service = IdentityEventService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetIdentityEventService"
+    )
+    protected void setIdentityEventService(IdentityEventService identityEventService) {
+
+        ApplicationManagementServiceComponentHolder.getInstance().setIdentityEventService(identityEventService);
+        if (log.isDebugEnabled()) {
+            log.debug("IdentityEventService set in Identity Claim Management bundle");
+        }
+    }
+
+    protected void unsetIdentityEventService(IdentityEventService identityEventService) {
+
+        ApplicationManagementServiceComponentHolder.getInstance().setIdentityEventService(null);
+        if (log.isDebugEnabled()) {
+            log.debug("IdentityEventService set in Identity Claim Management bundle");
+        }
```

Comment:
`if (log.isDebugEnabled())` check is not needed if the debug log is just a string.
internal code of log.debug uses the same check of isDebugEnabled.
If your debug log has a string concatenation kind of string processing , add the if check prior to the log in order to reduce the string manipulation cost.

Check line 473 as well

Reply:
Fixed